### PR TITLE
Migration of Mass and TorqueReaction postprocessors

### PR DESCRIPTION
**Pull request by sapitts**
**Relevant Design Information**
The mass post processor in this commit was migrated from the solid mechanics module into the tensor mechanics module.  
A new generalized torque post processor is added into tensor mechanics while the original solid mechanics capability is retained; the solid mechanics post processor included assumptions about the orientation of the force vector and the axis of rotation vector.  The new tensor mechanics torque reaction post processor uses a scalar algorithm to allow any orientation of rotation axis and force direction vectors.  The tensor mechanics torque reaction post processor is distinguished from the solid mechanic version with the word 'Scalar': TorqueReactionScalar.  

**Test Cases**
For the mass post processor, the tests for solid mechanics (stress divergence, ect) are preserved but moved to combined_modules, and identical tensor mechanics tests are added in the combined module; cli_args is employed to reduce the number of tests added.

In the case of the torque reaction post processor, the solid mechanics test is retained to provide code coverage for the original TorqueReaction post processor.  A new test is included for the tensor mechanics version.

Refs #6484

### CODE REVIEW

File: modules/tensor_mechanics/include/postprocessors/TorqueReactionScalar.h [hunk 1]
```diff
@@ -0,0 +1,65 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+
```
Comment (dschwen) [inline modules/tensor_mechanics/include/postprocessors/TorqueReactionScalar.h]: remove

File: modules/tensor_mechanics/include/postprocessors/TorqueReactionScalar.h [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+ * acting torques by projecting both the reaction force and the position vector
+ * (the coordinates of the node upon which the force is applied) onto the axis of
+ * rotation and applying the Pythagorean theorem, as in a statics course.  This
+ * scalar approach allows the postprocessor to accept any axis of rotation direction.
+ *
+ * TorqueReactionScalarScalar is similar to TorqueReactionScalar in SolidMechanics but does
+ * not replace the TorqueReactionScalar postprocessor; different assumptions were used
+ * to derive the SolidMechanics TorqueReactionScalar postprocessor.
+ */
+
```
Comment (dschwen) [inline modules/tensor_mechanics/include/postprocessors/TorqueReactionScalar.h]: remove (and the line after `protected:` as well)

File: modules/tensor_mechanics/src/base/TensorMechanicsApp.C [hunk 3]
```diff
@@ -234,6 +237,9 @@ TensorMechanicsApp::registerObjects(Factory & factory)
   registerAux(NewmarkVelAux);
 
   registerBoundaryCondition(Pressure);
+
+  registerPostprocessor(Mass);
```
Comment (dschwen) [inline modules/tensor_mechanics/src/base/TensorMechanicsApp.C]: I guess we should add a `registerNodalPostprocessor` macro in `Factory.h`...

File: modules/tensor_mechanics/src/postprocessors/TorqueReactionScalar.C [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+/****************************************************************/
+
+
+#include "TorqueReactionScalar.h"
+
+template<>
+InputParameters validParams<TorqueReactionScalar>()
+{
+  InputParameters params = validParams<NodalPostprocessor>();
+  params.addParam<AuxVariableName>("react_x","The x reaction variable");
```
Comment (dschwen) [inline modules/tensor_mechanics/src/postprocessors/TorqueReactionScalar.C]: space after `,` and no double empty lines up top.

File: modules/tensor_mechanics/src/postprocessors/TorqueReactionScalar.C [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  params.addParam<AuxVariableName>("react_y","The y reaction variable");
+  params.addParam<AuxVariableName>("react_z","The z reaction variable");
+
+  params.addParam<RealVectorValue>("axis_origin", RealVectorValue(0., 0., 0.), "Origin of the axis of rotation used to calculate the torque");
```
Comment (dschwen) [inline modules/tensor_mechanics/src/postprocessors/TorqueReactionScalar.C]: just put `RealVectorValue()` or even better `Point()`

File: modules/tensor_mechanics/src/postprocessors/TorqueReactionScalar.C [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    _react_z(_react_z_var.nodalSln()),
+    _axis_origin(getParam<RealVectorValue>("axis_origin")),
+    _direction_vector(getParam<RealVectorValue>("direction_vector"))
+{
+}
+
+void
+TorqueReactionScalar::initialize()
+{
+  _sum = 0.;
```
Comment (dschwen) [inline modules/tensor_mechanics/src/postprocessors/TorqueReactionScalar.C]: I'm not a fan of the tuncated decimals. `0.0`

File: modules/tensor_mechanics/src/base/TensorMechanicsApp.C (hunk 3 above)
Comment (permcody) [inline modules/tensor_mechanics/src/base/TensorMechanicsApp.C]: Some of the devs have actually wanted to go away from more macros to where everything is simply `registerObject()`. I can see ups and downs. We should probably be consistent though but Stephanie doesn't need to do worry about it with this PR.

Comment (permcody) [inline modules/tensor_mechanics/src/base/TensorMechanicsApp.C]: Wow I can react to my own comment. I guess if I'm torn on something I can give myself a thumbs up and down.

**Comment by permcody** [comment]
@sapitts - torque_reaction.torque_reaction_tm is exodiffing on Intel:
https://www.moosebuild.org/job/17054/

**Comment by sapitts** [comment]
Thanks @permcody, looks like an error in my math somewhere because the post processor values are off.  Can I push to this branch to send the update or is it best to make a new branch for a new PR?

**Pull request merged**